runhouse/resources/functions/aws_lambda.py

```diff
@@ -0,0 +1,845 @@
+        LAMBDA_CLIENT = boto3.client("lambda")
+        LOGS_CLIENT = boto3.client("logs")
+
+    else:
+        LAMBDA_CLIENT = boto3.client("lambda", region_name=DEFAULT_REGION)
+        LOGS_CLIENT = boto3.client("logs", region_name=DEFAULT_REGION)
+
+    def __init__(
+        self,
+        paths_to_code: list[str],
+        handler_function_name: str,
+        runtime: str,
+        args_names: list[str],
+        fn_pointers: tuple,
+        name: str,
+        env: Env,
+        timeout: int,  # seconds
+        memory_size: int,  # MB
+        tmp_size: int,  # MB
+        retention_time: int = DEFAULT_RETENTION,  # days
+        dryrun: bool = False,
+        access: Optional[str] = None,
+        **kwargs,  # We have this here to ignore extra arguments when calling from from_config
+    ):
+        """
+        Runhouse AWS Lambda object. It is comprised of the entry point, configuration,
+        and dependencies necessary to run the service.
+
+        .. note::
+                To create an AWS lambda resource, please use the factory method :func:`aws_lambda_function`.
+        """
+
+        super().__init__(
+            name=name,
+            dryrun=dryrun,
+            system=self.RESOURCE_TYPE,
+            env=env,
+            fn_pointers=fn_pointers,
+            access=access or self.DEFAULT_ACCESS,
+            **kwargs,
+        )
+
+        self.local_path_to_code = paths_to_code
+        self.handler_function_name = handler_function_name
+        self.args_names = args_names
+        self.runtime = runtime
+        self.timeout = timeout
+        self.memory_size = memory_size
+        self.tmp_size = tmp_size
+        self.aws_lambda_config = (
+            None  # Lambda config and role arn from aws will be saved here
+        )
+        self.retention_time = retention_time
+
+    # ----------------- Constructor helper methods -----------------
+
+    @classmethod
+    def from_config(cls, config: dict, dryrun: bool = False):
+        """Create an AWS Lambda object from a config dictionary."""
+
+        if "resource_subtype" in config.keys():
+            config.pop("resource_subtype", None)
+        if "system" in config.keys():
+            config.pop("system", None)
+        keys = config.keys()
+        if "fn_pointers" not in keys:
+            config["fn_pointers"] = None
+        if "timeout" not in keys:
+            config["timeout"] = AWSLambdaFunction.DEFAULT_TIMEOUT
+        if "memory_size" not in keys:
+            config["memory_size"] = AWSLambdaFunction.DEFAULT_MEMORY_SIZE
+        if "env" not in keys:
+            config["env"] = Env(
+                reqs=[], env_vars={"HOME": "/tmp"}, name=Env.DEFAULT_NAME
+            )
+        else:
+            config["env"] = Env.from_config(config["env"])
+
+        return AWSLambdaFunction(**config, dryrun=dryrun).to()
+
+    @classmethod
+    def from_name(cls, name, dryrun=False, alt_options=None):
+        funcs = cls.LAMBDA_CLIENT.list_functions()
```

Comment:
Expanding here on my previous comment, but another reason to have this load from Den and not the lambda client is to be able to support reloading shared functions.

This also means that we need to parse `name` here, since it can represent an rns address